README.md

```diff
@@ -1,43 +1,466 @@
+    // :snippet-end:
+  }
+}
+```
+
+Running the following command:
+
+```
+bluehawk snip Main.java -d . --state subtract-one
+```
+
+Produces the following output:
+
+`Main.codeblock.add-or-subtract.java`:
+
+```java
+    int example = 1;
+    example--;
+    System.out.println("Example: " + example);
+```
+
+> 💡 Note that Bluehawk has trimmed one layer of comments from the `hello-user`
+> state in the produced code block.
+
+With `state-uncomment`, you can create multiple valid end states but only run
+one of those states when executing your source code.
+
+### Uncomment
+
+The `uncomment` command removes a single comment from the beginning of
+each line of the spanned range in all output.
+Because `uncomment` operates on ranges of content, it is only available as
+a block command.
+
+Consider the following file:
+
+`Main.java`:
+
+```java
+public class Main {
+  public static void main(String[] args){
+    int example = 1;
+    // :uncomment-start:
+    //example--;
+    // :uncomment-end:
+    example++;
+    System.out.println("Example: " + example);
+  }
+}
+```
+
+Running the following command:
+
+```
+bluehawk copy Main.java -d .
+```
+
+Produces the following output:
+
+`Main.java`:
+
+```java
+public class Main {
+  public static void main(String[] args){
+    int example = 1;
+    example--;
+    example++;
+    System.out.println("Example: " + example);
+  }
+}
+```
+
+### Replace
+
+The `replace` command accepts a JSON dictionary called "terms" as input,
+and replaces occurrences string keys in the map within the
+spanned range with their map values in all output. You can use
+`replace` to hide implementation details like complicated class names
+or API endpoint URLs in generated output. Because `replace` operates
+on ranges of content, it is only available as a block command.
+
+Consider the following file:
+
+`Main.java`:
+
+```java
+// :replace-start: {
+//    "terms": {
+//       "MyMainExample": "Main",
+//       "www.example.com/rest/v1": "YOUR_REST_ENDPOINT_HERE"
+//    }
+// }
+
+/*
+ * MyMainExample -- a class that contains only a hello world main method
+ * that defines a rest endpoint.
+ */
+public class MyMainExample {
+  String rest_endpoint;
+
+  public static void main(String[] args){
+    System.out.println("Hello world!");
+    rest_endpoint = "www.example.com/rest/v1"
+  }
+}
+// :replace-end:
+```
+
+Running the following command:
+
+```
+bluehawk copy Main.java -d .
+```
+
+Produces the following output:
+
+`Main.java`:
+
+```java
+/*
+ * Main -- a class that contains only a hello world main method.
+ */
+public class Main {
+  String rest_endpoint;
+
+  public static void main(String[] args){
+    System.out.println("Hello world!");
+    rest_endpoint = "YOUR_REST_ENDPOINT_HERE"
+  }
+}
+```
+
+### Emphasize
+
+The `emphasize` command only applies to [formatted](#format)
+output. When the `--format` flag is not used to generate formatted
+output, `emphasize` is discarded completely. When the `--format` flag
```

Comment:
This paragraph could use some work. I think it could be tightened up a bit. The "is not used... is discarded" is totally passive voice and repeating information already in the paragraph. So, maybe it would be clearer to say something like:

>... What the command does...
>
>NOTE: The `emphasize` command only applies to formatted output. *You must use the `--format` flag with the CLI to get formatted output.*